lib/auth.ts

```diff
@@ -520,18 +520,30 @@ const getAuth = function (
 		const email = verificationPayload.email;
 		const verificationToken = verificationPayload.token;
 
-		const { body } = await request.send({
-			method: 'POST',
-			url: '/user/v1/verify-email',
-			body: {
-				verificationToken,
-				email,
-			},
-			baseUrl: apiUrl,
-			sendToken: false,
-		});
+		try {
+			const { body } = await request.send({
+				method: 'POST',
+				url: '/user/v1/verify-email',
+				body: {
+					verificationToken,
+					email,
+				},
+				baseUrl: apiUrl,
+				sendToken: false,
+			});
 
-		return body;
+			return body;
+		} catch (err) {
+			if (err.statusCode === 400) {
+				throw new errors.BalenaRequestError(
+					'Your verification link has expired - log in again to request another one.',
```

Comment:
I took a better look of our messaging and flow and I'm still not convinced that the SDK is the right place for this.
Especially since `link` and `log in again` refer to a UI specific user flow that the SDK doesn't necessarily care about (the sdk really just takes a code and doesn't require you to re-login if you are already logged-in in order to request a new code).

What if we later want to add a "request new code" in the same page that the user verification link navigates you to, thus the user not having to re-login to request a new one if expired?
Doing such a UI-only user flow change should not also require changes to the SDK code right (the error messaging in this case)?

Imo let's do the simplest thing of having UI-specific user messaging be in the UI.

Reply:
@thgreasi Didn't quite get the scenario you mentioned below:
> What if we later want to add a "request new code" in the same page that the user verification link navigates you to, thus the user not having to re-login to request a new one if expired?
Doing such a UI-only user flow change should not also require changes to the SDK code right (the error messaging in this case)?

Also, I agree to having this part of UI as this is only UI specific messaging. Do we want to segregate the type of error on the UI or update the wording of the above error msg to be a bit more generic? e.g. `'Issue with the verification link provided - log in again to generate a new one.'`